src/lib/utils/options/options.ts

```diff
@@ -320,6 +314,22 @@ export class Options {
         }
         return errors.length ? Err(errors) : Ok(void 0);
     }
+
+    /**
+     * Sets the value of a given option to its default value.
+     * @param declaration The option whoes value should be reset.
+     */
+    private setOptionValueToDefault(declaration: Readonly<DeclarationOption>): void {
+        if (declaration.scope !== ParameterScope.TypeScript) {
+            // No nead to convert the defaultValue for a map type as it has to be of a specific type
+            if (declaration.type === ParameterType.Map) {
```

Comment:
Could you add a test for your specific usage that illustrates why this is important? The one you posted in gitter should be fine, or even something simpler.

Reply:
Good idea. Test added. I ran it through. Previously all but the new test passed. Now all tests pass.